### PR TITLE
Bring long haul tests to v2

### PR DESCRIPTION
**Pull request by brycewang-microsoft**

### CODE REVIEW

File: e2e/LongHaul/.gitignore
Comment (drwill-ms) [inline e2e/LongHaul/.gitignore:1]: We can get rid of this file now, actually.

File: e2e/LongHaul/device/SystemProperties.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using System.Runtime.InteropServices;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Azure.Devices.LongHaul.Device
+{
+    class SystemProperties
```
Comment (drwill-ms) [inline e2e/LongHaul/device/SystemProperties.cs:6]:
```suggestion
    internal class SystemProperties
```

**Review by drwill-ms — APPROVED** [review]

File: e2e/LongHaul/device/ExceptionExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Devices.LongHaul.Device
+{
+    internal static class ExceptionExtensions
+    {
+        internal static IEnumerable<Exception> Unwind(this Exception exception, bool unwindAggregate = false)
```
[the remark this answers is not in the document]
Comment (brycewang-microsoft) [reply e2e/LongHaul/device/ExceptionExtensions.cs:11]: True, deleted.

**Pull request merged**